syntax = "proto3";

package musicstreaming;

service MusicService {
  rpc ListUsuarios (Empty) returns (UsuariosResponse);
  rpc ListMusicas (Empty) returns (MusicasResponse);
  rpc ListPlaylistsDoUsuario (UsuarioIdRequest) returns (PlaylistsResponse);
  rpc ListMusicasDaPlaylist (PlaylistIdRequest) returns (MusicasResponse);
  rpc ListPlaylistsPorMusica (MusicaIdRequest) returns (PlaylistsResponse);
}

message Empty {}

message Usuario {
  int32 id = 1;
  string nome = 2;
  int32 idade = 3;
}

message Musica {
  int32 id = 1;
  string nome = 2;
  string artista = 3;
}

message Playlist {
  int32 id = 1;
  string nome = 2;
}

message UsuarioIdRequest {
  int32 usuario_id = 1;
}

message PlaylistIdRequest {
  int32 playlist_id = 1;
}

message MusicaIdRequest {
  int32 musica_id = 1;
}

message UsuariosResponse {
  repeated Usuario usuarios = 1;
}

message MusicasResponse {
  repeated Musica musicas = 1;
}

message PlaylistsResponse {
  repeated Playlist playlists = 1;
}
